modules/keymaps.nix

```diff
@@ -6,6 +6,65 @@
 with lib; let
   helpers = import ../lib/helpers.nix {inherit lib;};
 
+  # Generates maps for a lua config
+  genMaps = mode: maps: let
+    action2Config = action: {
```

Comment:
maybe we can do something like
```nix
action2Config = {
    action,
    silent ? false,
    expr ? false,
    desc ? "",
    <fill in other options>
} @ args: args
```

Then we can  call `action2config {inherit action;}` in case of string and `action2config action` in case of dict ? It avoids repeating the names twice